src/content/tutorial/react-step-3/react-step-3.mdx

```diff
@@ -301,233 +268,200 @@ const getRowItems = rows =>
+## Populate data table
 
-_Note: The Pagination component isn’t inherently connected in any way to the DataTable, we need to tell it what to do when a change occurs using the onChange prop. This includes both page size changes and displaying different rows._.
+Now that we have that data, let's populate the data table. Replace `console.log(organization);` with the following that destructures the `organization` object. Once we have the respositories, we can call our `getRowItems()` helper to build the data table rows.
 
-## Create component
+##### src/content/RepoPage/RepoPage.js
 
-Now we'll need to consume our fetched data. In React, we generally want to separate our data fetching logic from our presentational or visual content. We'll separate our visual content into the `RepoPage/RepoTable.js` file. It's important to keep in mind that because we only render our `RepoTable` when we have our data, we don't need to be as safe with `null`/`undefined` checking our individual row properties. We've established that they exist given the nature of our GraphQL query.
+```javascript
+// If we're here, we've got our data!
+const { repositories } = organization;
+const rows = getRowItems(repositories.nodes);
+```
 
-### Create file
+Then, towards the top of `RepoPage.js` delete the `rows` array because we no longer need the example rows.
 
-Go ahead and create a new `RepoTable.js` file in `src/content/RepoPage`.
+### Data table data
 
-### Imports
+The data table component and its pieces use a common React pattern called [render props](https://reactjs.org/docs/render-props.html). This a really powerful way for libraries to give developers control of rendering and manipulating their data.
+
+Revisiting `RepoTable.js`, we are already passing in our row objects along with headers for each column. The `render` prop is being used to tell the data table how to render the headers and rows. That prop takes a function that receives the processed headers and rows as arguments as well as some helper functions for rendering the table.
+
+One common hurdle with the data table is how to access data that might be not correspond with a table column but is needed to compute the value of a cell that does. The data table component processes and controls only the row properties which corresponds to headers (columns). Because of this, the `rows` object you get access to in the render prop function is _different_ than the one you passed in to the `rows` prop.
 
-Import the various components that we'll need to render our table at the top of `RepoTable.js`.
+We need to modify one apsect of the data table because if you expand a row, it says `Row description`. We want to update that with the repository description coming from the GitHub API. This is an example of where we need a simple look-up function to find the data we care about - data that does not directly correspond to a column.
+
```

Comment:
I think this rewrite does no longer makes it seem like this change solves the previous problem outlined in the previous component. It’s also jarring hopping to the other file without context.

Reply:
Now that we have GH data, we need to make one change to RepoTable.js to update "Row description". I attempted to set context on why we need to touch that file. Maybe updating heading from "Data table data" to "Render repository descriptions" will help.